### PR TITLE
lint: refactor to use common json parsing helper

### PR DESCRIPTION
**Pull request by ee7**
Something like this.

### CODE REVIEW

File: src/lint/validators.nim [hunk 1]
```diff
@@ -182,3 +182,20 @@ proc subdirsContain*(dir: string, files: openArray[string]): bool =
...
+  if fileExists(path):
+    let contents = readFile(path)
+    if not checkEmpty or not isEmptyOrWhitespace(contents):
+      try:
+        result = parseJson(newStringStream(contents), path)
+      except JsonParsingError:
+        b.setFalseAndPrint("JSON parsing error", getCurrentExceptionMsg())
+    else:
+      b.setFalseAndPrint("File is empty, but must contain at least the empty " &
+                         "array, `[]`", path)
```
Comment (ErikSchierboom) [inline src/lint/validators.nim:199]: The array suffix not necessarily the right message, as some JSON files should be empty objects then. Maybe just make it a bit more general about the file not being allowed to be empty?

**Review by ErikSchierboom — APPROVED** [review]
LGTM Just some tiny nits

**Comment by ee7** [comment]
Previously (with 8d7fd89 or ff5349b) this PR would cause the following incorrect diff to the `configlet lint` output:

#### clojure
```diff
+Missing file:
+./exercises/concept/interest-is-interesting/.meta/config.json
+
+Missing file:
+./exercises/concept/international-calling-connoisseur/.meta/config.json
+
+Missing file:
+./exercises/concept/squeaky-clean/.meta/config.json
+
```

#### dart
```diff
+Missing file:
+./exercises/concept/futures/.meta/config.json
+
+Missing file:
+./exercises/concept/numbers/.meta/config.json
+
+Missing file:
+./exercises/concept/strings/.meta/config.json
+
```

That is, we would warn a second time about missing files. It seems like a good idea to keep the missing file check in `parseJsonFile`, and not suppress it with some parameter. So let's check for the existence of JSON files only in the procs that validate them, which also allows us to simplify the linting success message.

With https://github.com/exercism/configlet/pull/228/commits/191a94d87c524a48677fbb92889da2007e741703, this PR now produces the below diff:
#### clojure
Some errors are moved until later:
```diff
-Missing file:
-./exercises/concept/interest-is-interesting/.meta/config.json
-
-Missing file:
-./exercises/concept/international-calling-connoisseur/.meta/config.json
-
-Missing file:
-./exercises/concept/squeaky-clean/.meta/config.json
-
+Missing file:
+./exercises/concept/interest-is-interesting/.meta/config.json
+
+Missing file:
+./exercises/concept/international-calling-connoisseur/.meta/config.json
+
+Missing file:
+./exercises/concept/squeaky-clean/.meta/config.json
+
```



#### swift
We fix a bug on `main`, where we warned about missing `links.json` files twice. This currently only affects `swift`.
```diff
-Missing file:
-./concepts/characters/links.json
-
-Missing file:
-./concepts/classes/links.json
-
-Missing file:
-./concepts/initializers/links.json
-
-Missing file:
-./concepts/opaque-indices/links.json
-
-Missing file:
-./concepts/strings/links.json
-
-Missing file:
-./concepts/structs/links.json
-
```

We were previously printing errors for missing `links.json` in both `isEveryConceptLinksFileValid` and `conceptFilesExist`:
https://github.com/exercism/configlet/blob/f24176d0cdc737e4686c526a81de0b26d11c0cb2/src/lint/concepts.nim#L65-L79

File: src/lint/validators.nim (hunk 1 above)
Comment (ee7) [reply src/lint/validators.nim]: Yeah, I know - but we don't have any JSON files where `{}` is valid.

I've tried to improve it in the [force-push diff](https://github.com/exercism/configlet/compare/8d7fd895f17ea16ece7a5d97743b6fd2e6dd4671..ff5349b828a5dbf32cbe2493c5ea10d85d3706e5) (a little noisy, sorry) - it allows us to print a special message for empty `links.json` files, while still printing a "file is empty" message for empty `.meta/config.json` files. Good enough for now?

**Review by ErikSchierboom — APPROVED** [review]
I'm fine with this for now. I still find the "allow empty array" slightly odd, but we can fix that later.

**Pull request merged**